bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5163,35 +5159,37 @@ String addSourceToArtifactAttribute(BlackboardAttribute attr, String source) thr
 	 *                          within tsk core \ref query_database_page
 	 */
 	public ArrayList<BlackboardArtifact> getMatchingArtifacts(String whereClause) throws TskCoreException {
-		CaseDbConnection connection = null;
-		Statement s = null;
-		ResultSet rs = null;
+		String query = "SELECT blackboard_artifacts.artifact_id AS artifact_id, "
```

Comment:
Can this method be deprecated with a referral to an equivalent Blackboard method?

Reply:
@rcordovano This one is tricky to say if we can deprecate. I am going to say maybe.

In the blackboard methods, the "Where" methods use an alias for the table names.  This method does not use tables alias, it uses the direct table name.  Therefore someone calling this method would pass in a where that refers to the table name i.e. blackboard_aritifacts.obj_id whereas the methods in blackboard expect the user to know the alias that is being used by those methods.

Reply:
Let's stay away from tricky right now. We will revisit this later in the context of the story about refactoring SleuthkitCase/Blackboard.